aptos-move/aptos-vm/src/move_vm_ext/session.rs

```diff
@@ -479,6 +479,13 @@ impl<'r, 'l> SessionExt<'r, 'l> {
             .filter(|(state_key, _)| !resource_write_set.contains_key(state_key))
             .collect();
 
+        let group_reads_needing_change = aggregator_change_set
+            .group_reads_needing_exchange
+            .into_iter()
+            .filter(|(state_key, _)| !resource_group_write_set.contains_key(state_key))
```

Comment:
I think you can use `filter_map` here

Reply:
but the map is again .cloned(), so we can do .filter(..).cloned().collect()